sys/kern/kern_sysctl.c

```diff
@@ -1846,11 +1846,11 @@ sysctl_old_kernel(struct sysctl_req *req, const void *p, size_t l)
 				memcpy_c((char * __capability)req->oldptr +
```

Comment:
I think this whole thing is outdated. We probably should use `memcpy_c` unconditionally. If we do something conditional it should be using `memcpynocap_c` by default.

Reply:
Not using `l` and `i` as variables might also be a good idea for another commit.

Reply:
I have the nocap change in the risc-v purecap kernel branch.  I will pull it up into dev next after this set of changes.   I thought i had already pulled it up but apparently didn't yet.

Reply:
The 'l' and 'i' are upstream, so that would be an upstream change, but perhaps not a bad change to make.